Sources/Async/Streams/ByteParserStream.swift

```diff
@@ -0,0 +1,181 @@
+public protocol ByteParserStream: Async.Stream, ConnectionContext where Input == UnsafeBufferPointer<UInt8> {
```

Comment:
last we discussed, i thought we were going to have these extend TranslatingStream